cross-chain/bob/contracts/canonical/L2TBTC.sol

```diff
@@ -0,0 +1,284 @@
+    /// @dev Requirements:
+    ///      - The caller must be the contract owner.
+    ///      - `guardian` must not be a guardian address already.
+    /// @param guardian The address to be added as a guardian.
+    function addGuardian(address guardian) external onlyOwner {
+        require(!isGuardian[guardian], "This address is already a guardian");
+        isGuardian[guardian] = true;
+        guardians.push(guardian);
+        emit GuardianAdded(guardian);
+    }
+
+    /// @notice Removes the address from the guardians list.
+    /// @dev Requirements:
+    ///      - The caller must be the contract owner.
+    ///      - `guardian` must be a guardian address.
+    /// @param guardian The address to be removed from the guardians list.
+    function removeGuardian(address guardian) external onlyOwner {
+        require(isGuardian[guardian], "This address is not a guardian");
+        delete isGuardian[guardian];
+
+        // We do not expect too many guardians so a simple loop is safe.
+        for (uint256 i = 0; i < guardians.length; i++) {
+            if (guardians[i] == guardian) {
+                guardians[i] = guardians[guardians.length - 1];
+                // slither-disable-next-line costly-loop
+                guardians.pop();
+                break;
+            }
+        }
+
+        emit GuardianRemoved(guardian);
+    }
+
+    /// @notice Allows the governance of the token contract to recover any ERC20
+    ///         sent mistakenly to the token contract address.
+    /// @param token The address of the token to be recovered.
+    /// @param recipient The token recipient address that will receive recovered
+    ///        tokens.
+    /// @param amount The amount to be recovered.
+    function recoverERC20(
+        IERC20Upgradeable token,
+        address recipient,
+        uint256 amount
+    ) external onlyOwner {
+        token.safeTransfer(recipient, amount);
+    }
+
+    /// @notice Allows the governance of the token contract to recover any
+    ///         ERC721 sent mistakenly to the token contract address.
+    /// @param token The address of the token to be recovered.
+    /// @param recipient The token recipient address that will receive the
+    ///        recovered token.
+    /// @param tokenId The ID of the ERC721 token to be recovered.
+    function recoverERC721(
+        IERC721Upgradeable token,
+        address recipient,
+        uint256 tokenId,
+        bytes calldata data
+    ) external onlyOwner {
+        token.safeTransferFrom(address(this), recipient, tokenId, data);
+    }
+
+    /// @notice Allows one of the guardians to pause mints and burns allowing
+    ///         avoidance of contagion in case of a chain- or bridge-specific
+    ///         incident.
+    /// @dev Requirements:
+    ///      - The caller must be a guardian.
+    ///      - The contract must not be already paused.
+    function pause() external onlyGuardian {
```

Comment:
I still dont understand the `pause() onlyGuardian` concept, does a single bizantine guardian (1 of 19) pausing the bridge make sense?
thanks for the patience here, but there guardians are the same of Wormhole network or are specific to this bridge?